docs/setup_vscode_macos.md

```diff
@@ -550,6 +550,21 @@ Now, running the debugger will not display the assembly file. However, it will n
 <img src="images/vscode_macos_038.png" width="768px" />
 
 
+### Debugging fails
+Sometimes on older Macs with an Intel chip or following an update, debugging in VS Code fails due to an incompatible bundled `debugserver` (bug report](https://github.com/vadimcn/codelldb/issues/999).
```

Comment:
@jamesjuett Do you have any more of an idea of how to tell when this is the problem?

Reply:
Not right now, but I'm planning to reach out to see if I can recruit a few students who have debugging issues on Mac to meet for a while and dig into this. Unfortunately I can't replicate the issue on my machine. However, I think it would be alright to go ahead and merge this PR in the meantime.

Deleting the `debugserver` bundled with CodeLLDB results in fallback to the system version, which will be present as long as students have CLI tools installed via `xcode-select --install`, which will be the case for anyone following our tutorials. (The only reason there's a bundled version is so that it could work without CLI tools installed.)

So, it should never hurt to delete it, which makes me comfortable recommending this as a troubleshooting step even without a diagnostic.

Reply:
I've come across this issue multiple times and the key giveaway is the message "Process exited with code -1" in the Debug Console tab. CodeLLDB always automatically switches to the Terminal tab (where there's no signs of anything wrong), so you usually have to click on the Debug Console tab manually and check the exit code.

[This](https://piazza.com/class/lqe53njwkqc44e/post/46) Piazza post from the current semester shows off the issue pretty well. There's a video attached and some screenshots in the followups.